File: src/packets/PacketS2CSourceInfo.js
const PacketS2C = require("./PacketS2C");

/**
 * Server -> Client
 * ID = 0x06
 * Zone Name
 * Sends source information
 * Data:
 *      (Unsigned Char) SourceID
 *      (String) Name
 *      (Unsigned Char) Source Type ID
 */
class PacketS2CSourceInfo extends PacketS2C {
    constructor(sourceID, name, sourceTypeID) {
        super();

        this._buffer.writeUInt8(sourceID);
        this._buffer.writeStringNT(name);
        this._buffer.writeUInt8(sourceTypeID);
    }

    getID() {
        return 0x06;
    }
}

module.exports = PacketS2CSourceInfo;

File: src/RNetPi.Core/Packets/PacketS2CSourceInfo.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x06
/// Source Info
/// Sends source information
/// Data:
///     (Unsigned Char) SourceID
///     (String) Name
///     (Unsigned Char) Source Type ID
/// </summary>
public class PacketS2CSourceInfo : PacketS2C
{
    public const byte ID = 0x06;

    public PacketS2CSourceInfo(byte sourceID, string name, byte sourceTypeID)
    {
        Writer.Write(sourceID);
        WriteNullTerminatedString(name);
        Writer.Write(sourceTypeID);
    }

    public override byte GetID() => ID;
}